apps/web/src/components/NavMenu/MainNavMenu.tsx

```diff
@@ -0,0 +1,203 @@
+/* eslint-disable react/forbid-elements */
+import { useEffect, useState } from "react";
+import XMarkIcon from "@heroicons/react/20/solid/XMarkIcon";
+import Bars3Icon from "@heroicons/react/24/solid/Bars3Icon";
+import { useIntl } from "react-intl";
+
+import { notEmpty, useIsSmallScreen } from "@gc-digital-talent/helpers";
+import {
+  getLocale,
+  localizePath,
+  oppositeLocale,
+  uiMessages,
+} from "@gc-digital-talent/i18n";
+import {
+  Button,
+  NavMenu,
+  NavMenuWrapper,
+  Separator,
+} from "@gc-digital-talent/ui";
+import {
+  ROLE_NAME,
+  useAuthentication,
+  useAuthorization,
+} from "@gc-digital-talent/auth";
+
+import NotificationDialog from "../NotificationDialog/NotificationDialog";
+import useNavContext from "../NavContext/useNavContext";
+import useMainNavLinks from "./useMainNavLinks";
+import ThemeSwitcher from "../ThemeSwitcher/ThemeSwitcher";
+import navMenuMessages from "./messages";
+
+const MainNavMenu = () => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+  const isSmallScreen = useIsSmallScreen(1080);
+
+  const changeToLang = oppositeLocale(locale);
+  const languageTogglePath = localizePath(location, changeToLang);
+
+  const { navRole } = useNavContext();
+  const { userAuthInfo } = useAuthorization();
+  const { loggedIn } = useAuthentication();
+
+  const { roleLinks, mainLinks, accountLinks, authLinks } = useMainNavLinks(
+    navRole,
+    loggedIn,
+    userAuthInfo?.roleAssignments?.filter(notEmpty) ?? [],
+  );
+  // retain menu preference in storage
+  const [isMenuOpen, setMenuOpen] = useState(false);
+  useEffect(() => {
+    if (isSmallScreen) {
+      setMenuOpen(false); // collapse menu if window resized to small
+    }
+  }, [isSmallScreen, setMenuOpen]);
+
+  const [isNotificationDialogOpen, setNotificationDialogOpen] = useState(false);
+
+  const roleAssignments = userAuthInfo?.roleAssignments
+    ?.filter(notEmpty)
+    .filter(
+      (roleAssignment) => roleAssignment.role?.name !== ROLE_NAME.BaseUser,
+    );
+
+  const roleNames: Record<string, string> = {
```

Comment:
You can simplify this a bit and tighten the typing at the same time.  You'll find a gap with "guest", too, if you do this!
![image](https://github.com/user-attachments/assets/6fd22903-9464-4cee-af76-96af10be2a5a)

Reply:
Same recommendation for the maps in apps/web/src/components/NavMenu/useMainNavLinks.tsx

Reply:
I was able to change [88c420b](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/88c420b6d4e479bcff054605bf700eea3809dbe5) the above. However, I'm not sure how it would work in `useMainNavLinks.tsx`, since the `role.name` returns a string in camal_case so the mapping doesn't work out.

![image](https://github.com/user-attachments/assets/567361b5-f49d-4a76-8370-f7d60d5595a4)

Reply:
You have to quote the property name, but it doesn't mind snake case.
![image](https://github.com/user-attachments/assets/2c98c422-af16-4f8e-802a-60d1a8e643d3)

Reply:
This would work, but we set our eslint rules to automatically remove the quotes on object keys when saving :laughing: I'm not sure where to go from here.